openmc/openmoc_compatible.py

```diff
@@ -0,0 +1,825 @@
+import copy
+import operator
+
+import numpy as np
+
+import openmoc
+import openmc
+import openmc.checkvalue as cv
+
+
+# A dictionary of all OpenMC Materials created
+# Keys   - Material IDs
+# Values - Materials
+OPENMC_MATERIALS = {}
+
+# A dictionary of all OpenMOC Materials created
+# Keys   - Material IDs
+# Values - Materials
+OPENMOC_MATERIALS = {}
+
+# A dictionary of all OpenMC Surfaces created
+# Keys   - Surface IDs
+# Values - Surfaces
+OPENMC_SURFACES = {}
+
+# A dictionary of all OpenMOC Surfaces created
+# Keys   - Surface IDs
+# Values - Surfaces
+OPENMOC_SURFACES = {}
+
+# A dictionary of all OpenMC Cells created
+# Keys   - Cell IDs
+# Values - Cells
+OPENMC_CELLS = {}
+
+# A dictionary of all OpenMOC Cells created
+# Keys   - Cell IDs
+# Values - Cells
+OPENMOC_CELLS = {}
+
+# A dictionary of all OpenMC Universes created
+# Keys   - Universes IDs
+# Values - Universes
+OPENMC_UNIVERSES = {}
+
+# A dictionary of all OpenMOC Universes created
+# Keys   - Universes IDs
+# Values - Universes
+OPENMOC_UNIVERSES = {}
+
+# A dictionary of all OpenMC Lattices created
+# Keys   - Lattice IDs
+# Values - Lattices
+OPENMC_LATTICES = {}
+
+# A dictionary of all OpenMOC Lattices created
+# Keys   - Lattice IDs
+# Values - Lattices
+OPENMOC_LATTICES = {}
+
+
+def get_openmoc_material(openmc_material):
+    """Return an OpenMOC material corresponding to an OpenMC material.
+
+    Parameters
+    ----------
+    openmc_material : openmc.Material
+        OpenMC material
+
+    Returns
+    -------
+    openmoc_material : openmoc.Material
+        Equivalent OpenMOC material
+
+    """
+
+    cv.check_type('openmc_material', openmc_material, openmc.Material)
+
+    material_id = openmc_material.id
+
+    # If this Material was already created, use it
+    if material_id in OPENMOC_MATERIALS:
+        return OPENMOC_MATERIALS[material_id]
+
+    # Create an OpenMOC Material to represent this OpenMC Material
+    name = str(openmc_material.name)
+    openmoc_material = openmoc.Material(id=material_id, name=name)
+
+    # Add the OpenMC Material to the global collection of all OpenMC Materials
+    OPENMC_MATERIALS[material_id] = openmc_material
+
+    # Add the OpenMOC Material to the global collection of all OpenMOC Materials
+    OPENMOC_MATERIALS[material_id] = openmoc_material
+
+    return openmoc_material
+
+
+def get_openmc_material(openmoc_material):
+    """Return an OpenMC material corresponding to an OpenMOC material.
+
+    Parameters
+    ----------
+    openmoc_material : openmoc.Material
+        OpenMOC material
+
+    Returns
+    -------
+    openmc_material : openmc.Material
+        Equivalent OpenMC material
+
+    """
+
+    cv.check_type('openmoc_material', openmoc_material, openmoc.Material)
+
+    material_id = openmoc_material.getId()
+
+    # If this Material was already created, use it
```

Comment:
This is copy/pasted directly from above, might be a good idea to make this a method and reuse it.